client/allocdir/alloc_dir.go

```diff
@@ -126,18 +126,24 @@ func (d *AllocDir) Embed(task string, dirs map[string]string) error {
 			continue
 		}
 
-		// Enumerate the files in source.
-		entries, err := ioutil.ReadDir(source)
-		if err != nil {
-			return fmt.Errorf("Couldn't read directory %v: %v", source, err)
+		// Check if destination directory exists. This can happen if restarting
+		// a failed task.
+		destDir := filepath.Join(taskdir, dest)
+		if _, err := os.Stat(destDir); err == nil {
```

Comment:
Yeah I can push this check down to the individual entries. Just need to avoid calling linkOrCopy on an existing link/copy as that will fail.